packages/ember/tests/component_registration_test.js

```diff
@@ -1,200 +1,172 @@
-import { jQuery } from 'ember-views';
-
-let App, appInstance;
-
-function prepare() {
-  setTemplate('components/expand-it', compile('<p>hello {{yield}}</p>'));
-  setTemplate('application', compile('Hello world {{#expand-it}}world{{/expand-it}}'));
-}
-
-function cleanup() {
-  run(() => {
-    try {
-      if (App) {
-        App.destroy();
-      }
-      App = appInstance = null;
-    } finally {
-      setTemplates({});
-    }
-  });
-}
-
-QUnit.module('Application Lifecycle - Component Registration', {
-  setup: prepare,
-  teardown: cleanup
-});
-
-function boot(callback, startURL = '/') {
-  run(() => {
-    App = Application.create({
-      name: 'App',
-      rootElement: '#qunit-fixture'
-    });
+import { Component } from 'ember-glimmer';
+import { moduleFor, ApplicationTestCase } from 'internal-test-helpers';
 
-    App.deferReadiness();
+moduleFor('Application Lifecycle - Component Registration', class extends ApplicationTestCase {
 
-    App.Router = Router.extend({
-      location: 'none'
+  ['@test The helper becomes the body of the component'](assert) {
+    this.addTemplate('application', 'Hello world {{#expand-it}}world{{/expand-it}}');
+    this.addComponent('expand-it', {
+      template: '<p>hello {{yield}}</p>'
     });
 
-    appInstance = App.__deprecatedInstance__;
-
-    if (callback) { callback(); }
-  });
-
-  let router = appInstance.lookup('router:main');
-
-  run(App, 'advanceReadiness');
-  run(() => router.handleURL(startURL));
-}
-
-QUnit.test('The helper becomes the body of the component', function() {
-  boot();
-  equal(jQuery('div.ember-view > div.ember-view', '#qunit-fixture').text(), 'hello world', 'The component is composed correctly');
-});
-
-QUnit.test('If a component is registered, it is used', function() {
-  boot(() => {
-    appInstance.register('component:expand-it', Component.extend({
-      classNames: 'testing123'
-    }));
-  });
+    this.visit('/').then(() => {
+      let text = this.$('div.ember-view > div.ember-view').text().trim();
+      assert.equal(text, 'hello world', 'The component is composed correctly');
+    });
+  }
+
+  ['@test If a component is registered, it is used'](assert) {
+    this.addTemplate('application', 'Hello world {{#expand-it}}world{{/expand-it}}');
+    this.addComponent('expand-it', {
+      ComponentClass: Component.extend({
+        classNames: 'testing123'
+      }),
+      template: '<p>hello {{yield}}</p>'
+    });
 
-  equal(jQuery('div.testing123', '#qunit-fixture').text(), 'hello world', 'The component is composed correctly');
-});
+    this.visit('/').then(() => {
+      let text = this.$('div.testing123').text().trim();
+      assert.equal(text, 'hello world', 'The component is composed correctly');
+    });
+  }
+
+  ['@test Late-registered components can be rendered with custom `layout` property'](assert) {
+    this.addTemplate('application', '<div id=\'wrapper\'>there goes {{my-hero}}</div>');
+    this.addComponent('my-hero', {
```

Comment:
per the original test, this should also be registered onto an autobooted application instance